In [5]:
import pandas as pd
import scipy.stats as stats

df = pd.read_csv("/content/randomized_teachers_rating_data.csv")

print(df.head())

     prof  Gender  Age Tenure  Evaluation  Students  Beauty CourseLevel
0  Prof52    Male   32    Yes        4.27        51    3.53       Lower
1  Prof93    Male   30    Yes        4.97        31    4.67       Upper
2  Prof15    Male   42     No        3.13        63    4.68       Lower
3  Prof72  Female   42     No        4.98        98    3.54       Upper
4  Prof61  Female   44    Yes        3.39        86    4.01       Lower


# Q1. T-Test: Using the teachers' rating data set, does gender affect teaching evaluation rates?

In [6]:
correlation, p_value = stats.pearsonr(df['Beauty'], df['Evaluation'])

print(f"Correlation between Beauty and Evaluation: {correlation:.4f}")
print(f"P-value: {p_value:.4f}")

Correlation between Beauty and Evaluation: -0.0163
P-value: 0.8602


Correlation Coefficient: -0.0163

This value indicates a very weak negative linear relationship between Beauty and Evaluation.
In practical terms, this means that as a professor's perceived beauty score increases, their evaluation score tends to slightly decrease — but the effect is extremely small and negligible.

P-value: 0.8602

Since this p-value is greater than the standard significance level of 0.05, the correlation is not statistically significant.
This means we fail to reject the null hypothesis, indicating that the weak negative relationship observed in the sample could easily be due to random variation and does not provide strong evidence of a real relationship in the overall population.

# Q2. ANOVA: Using the teachers' rating data set, does beauty score for instructors differ by age?

## Objective:
To check whether beauty scores differ significantly across different age groups.

## Hypotheses:

H₀: Mean beauty scores are equal across all age groups.

H₁: At least one group has a different mean beauty score.

## Test Used:
One-way ANOVA

In [7]:
male_eval = df[df['Gender'] == 'Male']['Evaluation']
female_eval = df[df['Gender'] == 'Female']['Evaluation']

t_stat, p_value_gender = stats.ttest_ind(male_eval, female_eval)

print(f"T-statistic: {t_stat:.4f}")
print(f"P-value: {p_value_gender:.4f}")

T-statistic: -1.4065
P-value: 0.1622


T-statistic: -1.4065

This negative value indicates that the average 'Evaluation' score for male professors is slightly lower than the average score for female professors.
However, the difference is small and not practically meaningful.

P-value: 0.1622

Since this p-value is greater than the standard significance level of 0.05, the result is not statistically significant.
We therefore fail to reject the null hypothesis, indicating that there is no significant difference in the mean evaluation scores between male and female professors in this dataset.

# Q3. Chi-square: Using the teachers' rating data set, is there an association between tenure and gender?

# Objective:
To test whether tenure and gender are independent or associated.

# Hypotheses:

H₀: Tenure and gender are independent.

H₁: There is an association between tenure and gender.

Test Used:
Chi-square test of independence

In [8]:
contingency_table = pd.crosstab(df['Tenure'], df['Gender'])
chi2, p, dof, expected = stats.chi2_contingency(contingency_table)
print("Chi-square:", chi2)
print("P-Value:", p)

Chi-square: 0.3310169039069295
P-Value: 0.5650609108682598


# Interpretation:

fail to reject H₀ → They are independent.

# Q4. Correlation: Using the teachers' rating dataset, is teaching evaluation score correlated with beauty score?

# Objective:
To test whether there is a linear relationship between evaluation and beauty.

# Hypotheses:

H₀: There is no correlation between evaluation and beauty.

H₁: There is a correlation between evaluation and beauty.

Test Used:
Pearson Correlation Coefficient

In [9]:
tenured_eval = df[df['Tenure'] == 'Yes']['Evaluation']
non_tenured_eval = df[df['Tenure'] == 'No']['Evaluation']

t_stat_tenure, p_value_tenure = stats.ttest_ind(tenured_eval, non_tenured_eval)

print(f"T-statistic: {t_stat_tenure:.4f}")
print(f"P-value: {p_value_tenure:.4f}")

T-statistic: -0.1541
P-value: 0.8778


T-statistic: -0.1541

This negative value indicates that the average 'Evaluation' score for tenured professors is slightly lower than that of non-tenured professors.
However, the difference in means is very small and not practically meaningful.

P-value: 0.8778

Since this p-value is greater than the standard significance level of 0.05, the result is not statistically significant.
Therefore, we fail to reject the null hypothesis, indicating that tenure status does not have a significant effect on teaching evaluation scores.